contracts/core/exchange-wrappers/KyberNetworkWrapper.sol

```diff
@@ -0,0 +1,219 @@
+/*
+    Copyright 2018 Set Labs Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+pragma solidity 0.4.24;
+pragma experimental "ABIEncoderV2";
+
+import { SafeMath } from "zeppelin-solidity/contracts/math/SafeMath.sol";
+import { Authorizable } from "../../lib/Authorizable.sol";
+import { ERC20Wrapper as ERC20 } from "../../lib/ERC20Wrapper.sol";
+import { KyberNetworkProxyInterface } from "../../external/KyberNetwork/KyberNetworkProxyInterface.sol";
+import { LibBytes } from "../../external/0x/LibBytes.sol";
+
+
+/**
+ * @title KyberNetworkWrapper
+ * @author Set Protocol
+ *
+ * The KyberNetworkWrapper contract wrapper to interface with KyberNetwork for reserve liquidity
+ */
+contract KyberNetworkWrapper is
+    Authorizable
+{
+    using SafeMath for uint256;
+    using LibBytes for bytes;
+
+    /* ============ State Variables ============ */
+
+    address public kyberNetworkProxy;
+    address public setTransferProxy;
+
+    // ============ Structs ============
+
+    struct KyberTrade {
+        address sourceToken;
+        address destinationToken;
+        uint256 sourceTokenQuantity;
+        uint256 minimumConversionRate;
+        uint256 maxDestinationQuantity;
+    }
+
+    /* ============ Constructor ============ */
+
+    /**
+     * Initialize exchange wrapper with required addresses to facilitate Kyber trades
+     *
+     * @param  _kyberNetworkProxy    KyberNetwork contract for filling orders
+     * @param  _setTransferProxy     Set Protocol transfer proxy contract
+     */
+    constructor(
+        address _kyberNetworkProxy,
+        address _setTransferProxy
+    )
+        public
+        Authorizable(2592000) // about 4 weeks
+    {
+        kyberNetworkProxy = _kyberNetworkProxy;
+        setTransferProxy = _setTransferProxy;
+    }
+
+    /* ============ Public Functions ============ */
+
+    /**
+     * IExchangeWrapper interface delegate method.
+     * Parses and executes Kyber trades
+     *
+     * TODO: Currently passes change back to the issuance order maker
+     *
+     * @param  _maker         Unused address of issuance order signer to conform to IExchangeWrapper
```

Comment:
Comment is a little misleading because the address is used for transferring leftover tokens

Reply:
+1